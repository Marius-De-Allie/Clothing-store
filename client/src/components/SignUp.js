import React, {useState } from 'react';
import FormInput from './FormInput';
import CustomButton from './CustomButton';
import { auth, createUserProfileDocument } from '../firebase/firebase.utils';
// styles.
import '../styles/signUp.scss';

const SignUp = () => {
    const [credentials, setCredentials] = useState({
        displayName: '',
        email: '',
        password: '',
        confirmPassword: ''
    });

    const { displayName, email, password, confirmPassword } = credentials;

    const handleSubmit = async evt => {
        evt.preventDefault();

        if(password !== confirmPassword) {
            alert(`passwords don't match`);
            return;
        } else {
            try {
                const { user } = await auth.createUserWithEmailAndPassword(email, password);

                await createUserProfileDocument(user, {displayName});
                // empty input fields.
                setCredentials({
                    displayName: '',
                    email: '',
                    password: '',
                    confirmPassword: ''
                });
            } catch(error) {
                console.error(error);
            }
        }
    };
    
    const handleChange = evt => {
        const { name, value} = evt.target;
        setCredentials({...credentials, [name]: value});
    };
    
    return (
        <div className="sign-up">
            <h2 className="title">I do not have an account</h2>
            <span>Sign up with your email and password</span>
            <form className="sign-up-form" onSubmit={handleSubmit}>
                <FormInput 
                    type="text"
                    name="displayName"
                    value={displayName}
                    handleChange={handleChange}
                    label="Display Name"
                    required
                />
                <FormInput 
                    type="email"
                    name="email"
                    value={email}
                    handleChange={handleChange}
                    label="Email"
                    required
                />
                <FormInput 
                    type="password"
                    name="password"
                    value={password}
                    handleChange={handleChange}
                    label="Password"
                    required
                />
                <FormInput 
                    type="password"
                    name="confirmPassword"
                    value={confirmPassword}
                    handleChange={handleChange}
                    label="Confirm Password"
                    required
                />
                <CustomButton type='submit'>SIGN UP</CustomButton>
            </form>
        </div>
    );
};

export default SignUp;