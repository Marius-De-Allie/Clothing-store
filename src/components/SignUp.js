import React from 'react';
import FormInput from './FormInput';
import CustomButton from './CustomButton';
import { auth, createUserProfileDocument } from '../firebase/firebase.utils';
// styles.
import '../styles/signUp.scss';

class SignUp extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            displayName: '',
            email: '',
            password: '',
            confirmPassword: ''
        }
    }

    handleSubmit = async evt => {
        evt.preventDefault();

        const { displayName, email, password, confirmPassword } = this.state;
        
        if(password !== confirmPassword) {
            alert(`passwords don't match`);
            return;
        } else {
            try {
                const { user } = auth.createUserWithEmailAndPassword(email, password);

                await createUserProfileDocument(user, {displayName});
                // empty input fields.
                this.setState(() => ({
                    displayName: '',
                    email: '',
                    password: '',
                    confirmPassword: ''
                }));
            } catch(error) {
                console.error(error);
            }
        }
    };
    
    handleChange = evt => {
        const{ name, value} = evt.target;

        this.setState(() => ({
            [name]: value
        }));
    };

    render () {
        const { displayName, email, password, confirmPassword } = this.state;
        return (
            <div className="sign-up">
                <h2 className="title">I do not have an account</h2>
                <span>Sign up with your email and password</span>
                <form className="sign-up-form" onSubmit={this.handleSubmit}>
                    <FormInput 
                        type="text"
                        name="displayName"
                        value={displayName}
                        handleChange={this.handleChange}
                        lable="Display Name"
                        required
                    />
                    <FormInput 
                        type="email"
                        name="email"
                        value={email}
                        handleChange={this.handleChange}
                        lable="Email"
                        required
                    />
                    <FormInput 
                        type="password"
                        name="password"
                        value={password}
                        handleChange={this.handleChange}
                        lable="Password"
                        required
                    />
                    <FormInput 
                        type="password"
                        name="confirmPassword"
                        value={confirmPassword}
                        handleChange={this.handleChange}
                        lable="Confirm Password"
                        required
                    />
                    <CustomButton type='submit'>SIGN UP</CustomButton>
                </form>
            </div>
        );
    }
};

export default SignUp;