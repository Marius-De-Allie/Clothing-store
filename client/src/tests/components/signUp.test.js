import React from 'react';
import { shallow } from 'enzyme';
import SignUp from '../../components/SignUp';

it('SignUp component renders properly', () => {
    const wrapper = shallow(<SignUp />);
    expect(wrapper).toMatchSnapshot();
});