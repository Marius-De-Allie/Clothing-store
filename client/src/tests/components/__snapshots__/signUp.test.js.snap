// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp component renders properly 1`] = `ShallowWrapper {}`;
